### ISSUE TITLE
Sip NOTIFY

### ISSUE THREAD

**Issue by nick2601** [BelledonneCommunications/linphone-android#1300]
> > As per my understanding if the Call ID and Event from the second SUBSCRIBE request is matched correctly with the NOTIFY response going out from server , the Linphone Stack should process it .
> 
> Yes, but the To header of the NOTIFY doesn't match the FROM of the subscribe, hence the out of dialog.

@Viish But the **To** header of the NOTIFY matches with **Contact Address** of the SIP SUBSCRIBE 
```
SUBSCRIBE sip:mcptt-as-mcla-1.ims.mnc001.mcc001.3gppnetwork.org SIP/2.0
Via: SIP/2.0/UDP 172.27.232.105:44263;branch=z9hG4bK.MQV1wJ3DG;rport
From: <sip:mcx.test-111.iitb@ims.mnc001.mcc001.3gppnetwork.org>;tag=Ux9dIH7eh
To: sip:mcptt-as-mcla-1.ims.mnc001.mcc001.3gppnetwork.org
CSeq: 20 SUBSCRIBE
Call-ID: sVhwxkU8Xr
Max-Forwards: 70
Route: <sip:172.31.37.34:7684;lr>
Supported: replaces, outbound, gruu
Event: presence
Expires: 423197357
Content-Type: application/vnd.3gpp.mcptt-info+xml
Content-Length: 320
Contact: <sip:mcx.test-111.iitb@172.27.232.105:44263;transport=udp>;transport=udp;expires=600;+sip.instance="<urn:uuid:168e1fee-0a0f-0038-9ef0-cba44d3df8f7>"
P-Preferred-Identity: sip:mcx.test-111.iitbims.mnc001.mcc001.3gppnetwork.org
P-Preferred-Service: urn:urn-7:3gpp-service.ims.icsi.mcptt
P-Access-Network-Info: ADSL;utran-cell-id-3gpp=00000000
Privacy: none
Accept: application/pidf+xml
User-Agent: NonMCLinphoneAndroid/4.4.0 (Nokia 6.1 Plus) LinphoneSDK/4.5.0-beta.89+4aecab5-debug (pipelines/25895)

<?xml version="1.0" encoding="UTF-8"?>
<mcpttinfo xmlns="urn:3gpp:ns:mcpttInfo:1.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
<mcptt-Params>
<mcptt-request-uri type="Normal">
<mcpttURI>sip:mcptt.test-111.iitb@ims.mnc001.mcc001.3gppnetwork.org</mcpttURI>
</mcptt-request-uri>
</mcptt-Params>
</mcpttinfo>

2021-03-01 11:22:57:642 [org.linphone.debug/liblinphone] MESSAGE LinphoneEvent [0x72a5921c00] moving to subscription state LinphoneSubscriptionOutgoingProgress
2021-03-01 11:22:57:666 [org.linphone.debug/belle-sip] MESSAGE bellesip_wake_lock_acquire(): Android wake lock [belle-sip recv channel] acquired [ref=0x30f2]
2021-03-01 11:22:57:667 [org.linphone.debug/belle-sip] MESSAGE bellesip_wake_lock_acquire(): cast long of wakelock 12530
2021-03-01 11:22:57:667 [org.linphone.debug/belle-sip] MESSAGE channel [0x72a58b5f00]: starting recv background task with id=[30f2].
2021-03-01 11:22:57:667 [org.linphone.debug/belle-sip] MESSAGE channel [0x72a58b5f00]: received [386] new bytes from [UDP://172.31.37.34:7684]:
SIP/2.0 200 OK
Via: SIP/2.0/UDP 172.27.232.105:44263;branch=z9hG4bK.MQV1wJ3DG;rport=44263
From: <sip:mcx.test-111.iitb@ims.mnc001.mcc001.3gppnetwork.org>;tag=Ux9dIH7eh
To: <sip:mcptt-as-mcla-1.ims.mnc001.mcc001.3gppnetwork.org>;tag=NQNeScF4N15Xe
Call-ID: sVhwxkU8Xr
CSeq: 20 SUBSCRIBE
Contact: <sip:172.31.37.34:7684>
User-Agent: MCLabs Registration Server
Content-Length: 0

NOTIFY sip:mcx.test-111.iitb@172.27.232.105:44263;transport=udp SIP/2.0
Via: SIP/2.0/UDP 172.31.37.34:7684;rport;branch=z9hG4bKS8Xjgv9FDjSSQ
Max-Forwards: 70
From: <sip:mcptt-as-mcla-1.ims.mnc001.mcc001.3gppnetwork.org>;tag=p0e7t7Z7javga
To: <sip:mcx.test-111.iitb@172.27.232.105:44263>;transport=udp
Call-ID: sVhwxkU8Xr
CSeq: 32735195 NOTIFY
Accept-Contact: *;+g.3gpp.icsi-ref="urn%3Aurn-7%3A3gpp-service.ims.icsi.mcptt";require;explicit
User-Agent: MCLabs MCX Server
Event: presence
Subscription-State: active;expires=4294967295
Content-Type: application/pidf+xml
Content-Length: 264
P-Asserted-Service: urn:urn-7:3gpp-service.ims.icsi.mcptt
P-Asserted-Identity: <sip:mcptt-as-mcla-1.ims.mnc001.mcc001.3gppnetwork.or>

<?xml version="1.0" encoding="UTF-8"?><presence xmlns="urn:ietf:params:xml:ns:pidf" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:mcpttPI10="urn:3gpp:ns:mcpttPresInfo:1.0" entity="sip:mcptt.test-111.iitb@ims.mnc001.mcc001.3gppnetwork.org"></presence>
```
So as per RFC 3261 we can map NOTIFY request to the Contact address of Sip Subscribe Event ,so out of dialog notify should not occur.
So what exactly is the reason for the rejection of NOTIFY.

_Originally posted by @nick2601 in https://github.com/BelledonneCommunications/linphone-sdk/issues/111#issuecomment-787714167_

**Comment by Viish** [comment]
No.
If you take a look at https://tools.ietf.org/html/rfc3261#section-8.1.1.8, the contact address is only used to indicate to the server how to reach this particular device on the network. For example, if you have two devices with the same identity, each will have a different Contact header value.
You can also take a look at https://tools.ietf.org/html/rfc3261#section-12, that describe the ink between From/To headers in dialog & the URIs.

**Comment by nick2601** [comment]
```
2021-03-03 15:55:26.802 17083-17083/org.linphone.debug I/Linphone: NOTIFY sip:mcx.test-444.mcla@172.27.224.37:39200;transport=udp SIP/2.0
2021-03-03 15:55:26.802 17083-17083/org.linphone.debug I/Linphone: Via: SIP/2.0/UDP 172.31.37.34:8642;rport;branch=z9hG4bKem7gt9ym03XjS
2021-03-03 15:55:26.802 17083-17083/org.linphone.debug I/Linphone: Max-Forwards: 70
2021-03-03 15:55:26.802 17083-17083/org.linphone.debug I/Linphone: From: <sip:mcx.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org>;tag=a1v1BHXBvXKrB
2021-03-03 15:55:26.802 17083-17083/org.linphone.debug I/Linphone: To: <sip:mcx.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org>;tag=gYuannNJo
2021-03-03 15:55:26.802 17083-17083/org.linphone.debug I/Linphone: Call-ID: yY0cUfSD0G
2021-03-03 15:55:26.802 17083-17083/org.linphone.debug I/Linphone: CSeq: 32829771 NOTIFY
2021-03-03 15:55:26.802 17083-17083/org.linphone.debug I/Linphone: Contact: <sip:mcx.test-444.mcla@172.31.37.34:8642>
2021-03-03 15:55:26.802 17083-17083/org.linphone.debug I/Linphone: User-Agent: MCLabs MCX Server
2021-03-03 15:55:26.802 17083-17083/org.linphone.debug I/Linphone: Allow: INVITE, ACK, BYE, CANCEL, OPTIONS, MESSAGE, INFO, UPDATE, REGISTER, REFER, NOTIFY, PUBLISH, SUBSCRIBE
2021-03-03 15:55:26.802 17083-17083/org.linphone.debug I/Linphone: Supported: timer, path, replaces
2021-03-03 15:55:26.802 17083-17083/org.linphone.debug I/Linphone: Event: presence
2021-03-03 15:55:26.802 17083-17083/org.linphone.debug I/Linphone: Allow-Events: presence
2021-03-03 15:55:26.802 17083-17083/org.linphone.debug I/Linphone: Subscription-State: active;expires=4294967295
2021-03-03 15:55:26.803 17083-17083/org.linphone.debug I/Linphone: Content-Type: application/pidf+xml
2021-03-03 15:55:26.803 17083-17083/org.linphone.debug I/Linphone: Content-Length: 465
2021-03-03 15:55:26.803 17083-17083/org.linphone.debug I/Linphone: P-Asserted-Service: urn:urn-7:3gpp-service.ims.icsi.mcptt
2021-03-03 15:55:26.803 17083-17083/org.linphone.debug I/Linphone: P-Asserted-Identity: <sip:mcptt-server-orig-part.organization.org>
2021-03-03 15:55:26.803 17083-17083/org.linphone.debug I/Linphone: <?xml version="1.0" encoding="UTF-8"?>
2021-03-03 15:55:26.803 17083-17083/org.linphone.debug I/Linphone: <presence xmlns="urn:ietf:params:xml:ns:pidf" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:mcpttPI10="urn:3gpp:ns:mcpttPresInfo:1.0" entity="sip:mcptt.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org">
2021-03-03 15:55:26.803 17083-17083/org.linphone.debug I/Linphone: <tuple id="sip:mc.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org">
2021-03-03 15:55:26.803 17083-17083/org.linphone.debug I/Linphone: <status>
2021-03-03 15:55:26.803 17083-17083/org.linphone.debug I/Linphone: <mcpttPI10:affiliation group="test_group@organization.org" status="affiliated" expires="1614774326"/>
2021-03-03 15:55:26.803 17083-17083/org.linphone.debug I/Linphone: </status>
2021-03-03 15:55:26.803 17083-17083/org.linphone.debug I/Linphone: </tuple>
2021-03-03 15:55:26.803 17083-17083/org.linphone.debug I/Linphone: </presence>
2021-03-03 15:55:26.821 17083-17083/org.linphone.debug I/Linphone: channel [0x768ae00200] [865] bytes parsed
2021-03-03 15:55:26.821 17083-17083/org.linphone.debug I/Linphone: channel [0x768ae00200] read [465] bytes of body from [172.31.37.34:8642]
2021-03-03 15:55:26.821 17083-17083/org.linphone.debug W/Linphone: Receiving NOTIFY with to-tag but no know dialog here, rejecting
2021-03-03 15:55:26.822 17083-17083/org.linphone.debug I/Linphone: channel [0x768ae00200]: message sent to [UDP://172.31.37.34:8642], size: [321] bytes
    SIP/2.0 481 Call/transaction does not exist
    Via: SIP/2.0/UDP 172.31.37.34:8642;rport;branch=z9hG4bKem7gt9ym03XjS
    From: <sip:mcx.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org>;tag=a1v1BHXBvXKrB
    To: <sip:mcx.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org>;tag=gYuannNJo
    Call-ID: yY0cUfSD0G
    CSeq: 32829771 NOTIFY
```

@Viish My Notify is rejected ,Inspite of mapping the From Header of Subscribe to the To header of the NOTIFY, can you help me why it is happening 
Thanks in advance

**Comment by Viish** [comment]
Please attach log with the subscribe as well.

**Comment by nick2601** [comment]
> Please attach log with the subscribe as well.

[logs](https://www.linphone.org:444//tmp/603f6966738dc_4a8a8db5159f03a59869.gz)

**Comment by nick2601** [comment]
Hi @Viish any updates on this..

**Comment by Viish** [comment]
I can't tell you why, but indeed the first NOTIFY is rejected for some reason but not the following one:
```
2021-03-03 15:55:26:803 [org.linphone.debug/belle-sip] MESSAGE channel [0x768ae00200]: received [1330] new bytes from [UDP://172.31.37.34:8642]:
NOTIFY sip:mcx.test-444.mcla@172.27.224.37:39200;transport=udp SIP/2.0
Via: SIP/2.0/UDP 172.31.37.34:8642;rport;branch=z9hG4bKem7gt9ym03XjS
Max-Forwards: 70
From: <sip:mcx.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org>;tag=a1v1BHXBvXKrB
To: <sip:mcx.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org>;tag=gYuannNJo
Call-ID: yY0cUfSD0G
CSeq: 32829771 NOTIFY
Contact: <sip:mcx.test-444.mcla@172.31.37.34:8642>
User-Agent: MCLabs MCX Server
Allow: INVITE, ACK, BYE, CANCEL, OPTIONS, MESSAGE, INFO, UPDATE, REGISTER, REFER, NOTIFY, PUBLISH, SUBSCRIBE
Supported: timer, path, replaces
Event: presence
Allow-Events: presence
Subscription-State: active;expires=4294967295
Content-Type: application/pidf+xml
Content-Length: 465
P-Asserted-Service: urn:urn-7:3gpp-service.ims.icsi.mcptt
P-Asserted-Identity: <sip:mcptt-server-orig-part.organization.org>

<?xml version="1.0" encoding="UTF-8"?>
<presence xmlns="urn:ietf:params:xml:ns:pidf" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:mcpttPI10="urn:3gpp:ns:mcpttPresInfo:1.0" entity="sip:mcptt.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org">
<tuple id="sip:mc.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org">
<status>
<mcpttPI10:affiliation group="test_group@organization.org" status="affiliated" expires="1614774326"/>
</status>
</tuple>
</presence>

2021-03-03 15:55:26:821 [org.linphone.debug/belle-sip] MESSAGE channel [0x768ae00200] [865] bytes parsed
2021-03-03 15:55:26:821 [org.linphone.debug/belle-sip] MESSAGE channel [0x768ae00200] read [465] bytes of body from [172.31.37.34:8642]
2021-03-03 15:55:26:821 [org.linphone.debug/liblinphone] WARNING Receiving NOTIFY with to-tag but no know dialog here, rejecting
2021-03-03 15:55:26:822 [org.linphone.debug/belle-sip] MESSAGE channel [0x768ae00200]: message sent to [UDP://172.31.37.34:8642], size: [321] bytes
SIP/2.0 481 Call/transaction does not exist
Via: SIP/2.0/UDP 172.31.37.34:8642;rport;branch=z9hG4bKem7gt9ym03XjS
From: <sip:mcx.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org>;tag=a1v1BHXBvXKrB
To: <sip:mcx.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org>;tag=gYuannNJo
Call-ID: yY0cUfSD0G
CSeq: 32829771 NOTIFY


2021-03-03 15:55:26:822 [org.linphone.debug/belle-sip] MESSAGE channel [0x768ae00200]: ending recv background task with id=[32e6].
2021-03-03 15:55:26:823 [org.linphone.debug/belle-sip] MESSAGE bellesip_wake_lock_release(): Android wake lock released [ref=0x32e6]
2021-03-03 15:55:26:826 [org.linphone.debug/belle-sip] MESSAGE bellesip_wake_lock_acquire(): Android wake lock [belle-sip recv channel] acquired [ref=0x32f6]
2021-03-03 15:55:26:826 [org.linphone.debug/belle-sip] MESSAGE bellesip_wake_lock_acquire(): cast long of wakelock 13046
2021-03-03 15:55:26:826 [org.linphone.debug/belle-sip] MESSAGE channel [0x768ae00200]: starting recv background task with id=[32f6].
2021-03-03 15:55:26:827 [org.linphone.debug/belle-sip] MESSAGE channel [0x768ae00200]: received [1411] new bytes from [UDP://172.31.37.34:8642]:
NOTIFY sip:mcx.test-444.mcla@172.27.224.37:39200;transport=udp SIP/2.0
Via: SIP/2.0/UDP 172.31.37.34:8642;rport;branch=z9hG4bKFX09U4FrXcm5m
Route: <sip:172.27.224.37:39200>;transport=udp
Max-Forwards: 70
From: <sip:mcx.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org>;tag=gR64pMAMx5kk
To: <sip:mcx.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org>;tag=gYuannNJo
Call-ID: yY0cUfSD0G
CSeq: 265396305 NOTIFY
Contact: <sip:mcx.test-444.mcla@172.31.37.34:8642>
User-Agent: MCLabs MCX Server
Allow: INVITE, ACK, BYE, CANCEL, OPTIONS, MESSAGE, INFO, UPDATE, REGISTER, REFER, NOTIFY, PUBLISH, SUBSCRIBE
Supported: timer, path, replaces
Event: presence
Allow-Events: talk, hold, conference, presence, poc-settings, as-feature-event, dialog, line-seize, call-info, sla, include-session-description, presence.winfo, message-summary, refer
Subscription-State: active;expires=423197357
Content-Type: application/pidf+xml
Content-Length: 465

<?xml version="1.0" encoding="ISO-8859-1"?> 
<presence xmlns='urn:ietf:params:xml:ns:pidf' 
xmlns:dm='urn:ietf:params:xml:ns:pidf:data-model' 
xmlns:rpid='urn:ietf:params:xml:ns:pidf:rpid' 
xmlns:c='urn:ietf:params:xml:ns:pidf:cipid' entity='sip:mcx.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org'>
 <tuple id='t6a5ed77e'>
  <status>
   <basic>open</basic>
  </status>
 </tuple>
 <dm:person id='p06360c4a'>
  <dm:note>Available</dm:note>
 </dm:person>
</presence>
2021-03-03 15:55:26:843 [org.linphone.debug/belle-sip] MESSAGE channel [0x768ae00200] [946] bytes parsed
2021-03-03 15:55:26:843 [org.linphone.debug/belle-sip] MESSAGE channel [0x768ae00200] read [465] bytes of body from [172.31.37.34:8642]
2021-03-03 15:55:26:847 [org.linphone.debug/belle-sip] MESSAGE bellesip_wake_lock_acquire(): Android wake lock [belle-sip transaction(0x76216ba9a0)] acquired [ref=0x3316]
2021-03-03 15:55:26:847 [org.linphone.debug/belle-sip] MESSAGE bellesip_wake_lock_acquire(): cast long of wakelock 13078
2021-03-03 15:55:26:847 [org.linphone.debug/belle-sip] MESSAGE transaction [0x76216ba9a0]: starting transaction background task with id=[3316].
2021-03-03 15:55:26:847 [org.linphone.debug/belle-sip] MESSAGE Changing [server] [NOTIFY] transaction [0x76216ba9a0], from state [INIT] to [TRYING]
2021-03-03 15:55:26:849 [org.linphone.debug/belle-sip] MESSAGE Changing [server] [NOTIFY] transaction [0x76216ba9a0], from state [TRYING] to [COMPLETED]
2021-03-03 15:55:26:850 [org.linphone.debug/belle-sip] MESSAGE channel [0x768ae00200]: message sent to [UDP://172.31.37.34:8642], size: [444] bytes
SIP/2.0 200 Ok
Via: SIP/2.0/UDP 172.31.37.34:8642;rport;branch=z9hG4bKFX09U4FrXcm5m
From: <sip:mcx.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org>;tag=gR64pMAMx5kk
To: <sip:mcx.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org>;tag=gYuannNJo
Call-ID: yY0cUfSD0G
CSeq: 265396305 NOTIFY
User-Agent: NonMCLinphoneAndroid/4.4.0 (Nokia 6.1 Plus) LinphoneSDK/4.5.0-beta.92+b24928e-debug (pipelines/25991)
Supported: replaces, outbound, gruu
```

**Comment by Viish** [comment]
Actually it seems to be an issue with the TO header tag.
The 200 OK sent by the server after the SUBSCRIBE contains the following:
```
SIP/2.0 200 OK
Via: SIP/2.0/UDP 172.27.224.37:39200;branch=z9hG4bK.vqI94s-NU;rport=39200
From: <sip:mcx.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org>;tag=gYuannNJo
To: <sip:mcx.test-444.mcla@ims.mnc001.mcc001.3gppnetwork.org>;tag=gR64pMAMx5kk
```

If you check my above message logs, the second NOTIFY that is accepted as matching tags, not the other that is rejected.

**Comment by nick2601** [comment]
Does it have something to do with expire value in the incoming first Notify and stack is not able to process because its value is larger than the primitive int.

**Comment by nick2601** [comment]
Does it have something to do with expire value in the incoming first Notify and stack is not able to process because its value is larger than the primitive int.

**Comment by nick2601** [comment]
@Viish ,
1)There is some issue with the incoming NOTIFY response  , sometimes I am able to see it while sometimes it is lost ,what might the issue, I have verified that those packet are sent from the server but are not accepted by the device ,I am unable to figure this behaviour ,I am using the callback **onNotifyReceived()**, have you ever encountered this kind of situation.
I have also tried to change the transport from udp to tcp but this behaviour continues.

Screenshot for the same:
![image](https://user-images.githubusercontent.com/59530213/110628910-97caec00-81c9-11eb-858f-ab92fddabd39.png)
![image (1)](https://user-images.githubusercontent.com/59530213/110628935-a0232700-81c9-11eb-9aaa-527ff78eed5c.png)
![image (2)](https://user-images.githubusercontent.com/59530213/110628966-aa452580-81c9-11eb-85e6-2e6d8d7f1006.png)

2)I want to add value for the expires argument in **Event event = core.createPublish(address, "poc-settings", 429496729);**
As the function is accepting primitive int as an argument , I need to have the value to be around 10 digits,is there a way where I can add expires value around a 10 digit Integer value.

Logs for the same:
[logs](https://www.linphone.org:444//tmp/60489a1543450_e21830a8ee1b349cd62c.gz)
